SUASSystem/SUASSystem/image_processing.py

```diff
@@ -81,15 +82,26 @@ def run_autonomous_img_proc_process(logger_queue, location_log, interop_client_a
 
             drone_gps_location = location_log[closest_time_index]["current_location"]
             target_location = get_target_gps_location(target_map_center_pixel_coordinates, target_pixel_coordinates, drone_gps_location)
+            target_latitude = target_location.get_lat()
+            target_longitude = target_location.get_lon()
 
+            # Check if current target is already submitted
+            is_current_target_already_submitted = False
+            for index in range(len(submitted_target_locations)):
+                if [target_latitude, target_longitude] == submitted_target_locations[index]:
```

Comment:
This is a logic error. More likely than not, the locations will not be exact. You need to create a check based on a distance threshold.